File: backend/app/schemas/history.py
"""Schemas Pydantic para os endpoints de histórico."""

from datetime import datetime
from pydantic import BaseModel


class AnalysisHistoryItem(BaseModel):
    id: int
    timestamp: datetime
    predicted_class: str
    probability_attack: float
    risk_level: str
    inference_time_ms: float
    model_used: str
    explanation_text: str

    class Config:
        from_attributes = True  # permite montar o schema a partir do objeto SQLAlchemy


class AnalysisHistoryResponse(BaseModel):
    total: int
    items: list[AnalysisHistoryItem]

File: frontend/src/types/history.ts
export interface AnalysisHistoryItem {
  id: number;
  timestamp: string;
  predicted_class: string;
  probability_attack: number;
  risk_level: "Baixo" | "Médio" | "Alto" | "Crítico";
  inference_time_ms: number;
  model_used: string;
  explanation_text: string;
}

export interface AnalysisHistoryResponse {
  total: number;
  items: AnalysisHistoryItem[];
}

export interface HistoryFilters {
  predicted_class?: string;
  risk_level?: string;
  limit: number;
  offset: number;
}